lib/src/common/js_proxy.dart

```diff
@@ -145,11 +145,32 @@ dynamic dartValue(jsValue) {
       }
       _addDartInstance(jsValue, dartMap);
       return dartMap;
+    } else if (jsValue is JsFunction) {
+      return _dartType(jsValue);
```

Comment:
Oh sorry, I meant to ask: will there be need to add support for arbitrary closures later though?

For this specific use case of passing the types of properties, what you have seems good. I do wonder whether we'll need to support arbitrary closures in the future, and if so, I'm curious of what is the purpose as it may tell us whether using a wrapper like Jake suggested would work, or if we need to do some more sophisticated.